saleor/webhook/subscription_payload.py

```diff
@@ -0,0 +1,116 @@
+    except (ValueError, GraphQLSyntaxError):
+        return False
+    return True
+
+
+def initialize_context() -> HttpRequest:
+    """Prepare a request object for webhook subscription.
+
+    It creates a dummy request object and initialize middleware on it. It is required
+    to process a request in the same way as API logic does.
+    return: HttpRequest
+    """
+    handler = BaseHandler()
+    context = RequestFactory().request()
+    handler.load_middleware()
+    handler.get_response(context)
+    return context
+
+
+def generate_payload_from_subscription(
+    event_type: str,
+    subscribable_object,
+    subscription_query: Optional[str],
+    context: HttpRequest,
+    app: Optional[App] = None,
+) -> Optional[Dict[str, Any]]:
+    """Generate webhook payload from subscription query.
+
+    It uses a graphql's engine to build payload by using the same logic as response.
+    As an input it expects given event type and object and the query which will be
+    used to resolve a payload.
+    event_type: is a event which will be triggered.
+    subscribable_object: is a object which have a dedicated own type in Subscription
+    definition.
+    subscription_query: query used to prepare a payload via graphql engine.
+    context: A dummy request used to share context between apps in order to use
+    dataloaders benefits.
+    app: the owner of the given payload. Required in case when webhook contains
+    protected fields.
+    return: A payload ready to send via webhook. None if the function was not able to
+    generate a payload
+    """
+    from ..graphql.api import schema
+
+    graphql_backend = get_default_backend()
+    ast = parse(subscription_query)  # type: ignore
+    document = graphql_backend.document_from_string(
+        schema,
+        ast,
+    )
+    graphql_middleware = graphene_settings.MIDDLEWARE
+    graphql_middleware = MiddlewareManager(
+        instantiate_middleware(graphql_middleware), wrap_in_promise=False
+    )
+    app_id = app.pk if app else None
+
+    context.app = app  # type: ignore
+
+    results = document.execute(
+        allow_subscriptions=True,
+        root=(event_type, subscribable_object),
+        context=context,
+        middleware=graphql_middleware,
+    )
+    if hasattr(results, "errors"):
+        logger.warning(str(results.errors))
+        logger.warning(
+            "Unable to build a payload for subscription",
+            extra={"query": subscription_query, "app": app_id},
+        )
+        return None
+    payload = []  # type: ignore
+    results.subscribe(payload.append)
+
+    if not payload:
+        logger.warning(
+            "Subscription did not return a payload.",
+            extra={"query": subscription_query, "app": app_id},
+        )
+
+    payload_instance = payload[0]
+
+    event_payload = payload_instance.data.get("event")
+
+    # TODO Make sure that we handle missing permissions in proper way.
```

Comment:
Should we resolve it or remove it?

Reply:
we decided to remove it from MVP, added a task for to deal with it in the future development of this feature